test/helpers/story/drivers/dapp-utils/compound.utils.ts

```diff
@@ -0,0 +1,143 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import chai, { expect } from 'chai'
+import { solidity } from 'ethereum-waffle'
+import { HardhatRuntimeEnvironment } from "hardhat/types"
+
+import { getDappAddresses } from '../../../../../config'
+import { ICErc20, IComptroller } from "../../../../../types/typechain"
+import { getFunds } from '../../../get-funds'
+import { LoanHelpersReturn } from "../../../loans"
+import LoanStoryTestDriver from '../loan-story-test-driver'
+chai.should()
+chai.use(solidity)
+
+async function lendCompound(
+    hre: HardhatRuntimeEnvironment,
+    loan: LoanHelpersReturn
+): Promise<void> {
+    const {contracts} = hre
+    const { details, diamond } = loan
+    await details.lendingToken
+      .connect(details.borrower.signer)
+      .approve(
+        diamond.address,
+        BigNumber.from(details.loan.borrowedAmount).mul(2)
+      )
+    const cToken = await contracts.get<ICErc20>('ICErc20', {
+      at: await diamond.getAssetCToken(details.lendingToken.address),
+    })
+    await diamond
+      .connect(details.borrower.signer)
+      .compoundLend(
+        details.loan.id,
+        details.loan.lendingToken,
+        details.loan.borrowedAmount
+      )
+    const escrowAddress = await diamond.getLoanEscrow(details.loan.id)
+    const cDaiBalance = await cToken.balanceOf(escrowAddress)
+    cDaiBalance.eq(0).should.eql(false, '')
+
+    const tokenAddresses = await diamond.getEscrowTokens(details.loan.id)
+    tokenAddresses.should.include(cToken.address)
+  }
+
+  async function withdrawCompound(
+    hre: HardhatRuntimeEnvironment,
+    loan: LoanHelpersReturn
+  ): Promise<void> {
+    const {contracts} = hre
+    const { details, diamond } = loan
+    await details.lendingToken
+      .connect(details.borrower.signer)
+      .approve(
+        diamond.address,
+        BigNumber.from(details.loan.borrowedAmount).mul(2)
+      )
+    const cToken = await contracts.get<ICErc20>('ICErc20', {
+      at: await diamond.getAssetCToken(details.lendingToken.address),
+    })
+    await diamond
+      .connect(details.borrower.signer)
+      .compoundRedeemAll(
+        details.loan.id,
+        details.lendingToken.address,
+      )
+    const escrowAddress = await diamond.getLoanEscrow(details.loan.id)
+
+    const tokenAddresses = await diamond.getEscrowTokens(details.loan.id)
+    tokenAddresses.should.not.include(cToken.address)
+
+    const cDaiBalance = await cToken.balanceOf(escrowAddress)
+    cDaiBalance.eq(0).should.eql(true, '')
```

Comment:
@dumebi not sure what the '' is for but maybe add a custom error message so when something fails we be like oooo dis balance should be equal but it ain't... ya know?